### PR TITLE
PageRank performance benchmark comparing Cassovary and GraphJet

### PR DESCRIPTION
**Pull request by lintool**
Benchmarks PageRank on a GraphJet graph vs. PageRank on a Cassovary graph.

### CODE REVIEW

File: graphjet-demo/src/main/java/com/twitter/graphjet/demo/PageRankGraphJetDemo.java [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+ */
+public class PageRankGraphJetDemo {
+  private static class PageRankGraphJetDemoArgs {
+    @Option(name = "-inputFile", metaVar = "[value]",
+        usage = "input data", required = true)
+    String inputFile;
+
+    @Option(name = "-maxSegments", metaVar = "[value]",
+        usage = "maximum number of segments")
+    int maxSegments = 20;
```
Comment (aneeshs) [inline graphjet-demo/src/main/java/com/twitter/graphjet/demo/PageRankGraphJetDemo.java]: note that reducing #segments and increasing edges/segment will likely help boost performance --- in the limit, a single segment should not be too much slower than cassovary (it still does a number of hash lookups)

File: graphjet-demo/src/main/java/com/twitter/graphjet/demo/PageRankGraphJetDemo.java [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+          BufferedReader br = new BufferedReader(new InputStreamReader(gzip));
+          String line;
+          while((line = br.readLine()) != null) {
+            if (line.startsWith("#")) continue;
+
+            String[] tokens = line.split("\\s+");
+            if (tokens.length > 1) {
+              final long from = Long.parseLong(tokens[0]);
+              final long to = Long.parseLong(tokens[1]);
+              graph.addEdge(from, to, (byte) 1);
```
Comment (aneeshs) [inline graphjet-demo/src/main/java/com/twitter/graphjet/demo/PageRankGraphJetDemo.java]: can store a reuse a "(byte) 1" variable

File: graphjet-demo/src/main/java/com/twitter/graphjet/demo/PageRankGraphJetDemo.java (hunk 1 above)
Comment (lintool) [inline graphjet-demo/src/main/java/com/twitter/graphjet/demo/PageRankGraphJetDemo.java]: Yes, once I get everything merged to master I will start playing with the parameters.

File: graphjet-demo/src/main/java/com/twitter/graphjet/demo/PageRankGraphJetDemo.java (hunk 2 above)
Comment (lintool) [inline graphjet-demo/src/main/java/com/twitter/graphjet/demo/PageRankGraphJetDemo.java]: done.

File: graphjet-demo/src/main/java/com/twitter/graphjet/demo/TopNodes.java [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+  }
+
+  /**
+   * Returns the top <i>k</i> nodes encountered by this heap.
+   *
+   * @return the top <i>k</i> nodes encountered by this heap.
+   */
+  public List<NodeValueEntry> getNodes() {
+    NodeValueEntry e;
+    List<NodeValueEntry> entries = new ArrayList<>(queue.size());
```
Comment (aneeshs) [inline graphjet-demo/src/main/java/com/twitter/graphjet/demo/TopNodes.java]: Note that if a client reuses this function, the behavior is somewhat strange: the top-k items are already pulled out, but some of them could be reinserted so at that point it won't be top (k+1, 2k) either. If you'd like to make this reusable, then you'd probably want to insert the top-k items back in the heap. And possibly think of synchronization.

Comment (aneeshs) [inline graphjet-demo/src/main/java/com/twitter/graphjet/demo/TopNodes.java]: Sorry, I just noticed that you actually empty out the queue. In that case, the behavior is easier to define but still keep some of the above in mind...

**Pull request merged**